workflow/controller/dag.go

```diff
@@ -140,15 +146,24 @@ func (d *dagContext) assessDAGPhase(targetTasks []string, nodes wfv1.Nodes) wfv1
 		targetTaskPhases[d.taskNodeID(task)] = ""
 	}
 
+	boundaryNode, err := nodes.Get(d.boundaryID)
```

Comment:
for context for other readers, the `nodes.Get` modifications here are just error guards a la #11451. no real semantic changes